src/main/java/de/tum/in/www1/artemis/web/rest/StatisticsResource.java

```diff
@@ -8,14 +8,15 @@
 
 import de.tum.in.www1.artemis.domain.enumeration.GraphType;
 import de.tum.in.www1.artemis.domain.enumeration.SpanType;
-import de.tum.in.www1.artemis.service.StatisticsService;
+import de.tum.in.www1.artemis.service.*;
+import de.tum.in.www1.artemis.web.rest.dto.CourseManagementStatisticsDTO;
 
 /**
- * REST controller for managing user statistics.
+ * REST controller for managing statistics.
  */
 @RestController
 @RequestMapping("/api")
-@PreAuthorize("hasRole('ADMIN')")
+@PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
```

Comment:
This change looks wrong. But should rather be `hasRole('TA')`.

Reply:
It actually made sense since the StatisticsResource was only used for the user statistics and therefore only for administrators. Since we also use the Resource for the Course statistics now, also TA und Instructors need to have permissions. With the new `hasRole` implementation by Stephan, however, we can use `hasRole('TA')` 👍